crapssim/bet.py

```diff
@@ -550,3 +550,59 @@ def _update_bet(self, table_object: "Table", dice_object: Dice) -> tuple[str | N
 
     def allowed(self, table: 'Table') -> bool:
         return table.new_shooter
+
+
+class AllBet(Bet):
+    def __init__(self, bet_amount: float):
+        super().__init__(bet_amount)
+        self.numbers = []
+        self.rolled_numbers: list[int] = []
+
+    def _update_bet(self, table_object: "Table", dice_object: Dice) -> tuple[str | None, float, bool]:
+        status: str | None = None
+        win_amount: float = 0.0
+        remove = False
+
+        if dice_object.total in self.numbers and dice_object.total not in self.rolled_numbers:
+            self.rolled_numbers.append(dice_object.total)
+            self.rolled_numbers.sort()
+
+        if self.numbers == self.rolled_numbers:
+            status = 'win'
+            win_amount = self.bet_amount * self.payoutratio
+            remove = True
+        elif dice_object.total == 7:
+            status = 'lose'
+            remove = True
+
+        return status, win_amount, remove
+
+    def allowed(self, table: 'Table') -> bool:
+        return table.new_shooter
+
+
+class AllTall(AllBet):
+    def __init__(self, bet_amount):
+        super().__init__(bet_amount)
+        self.name = 'AllTall'
+        self.bet_amount = bet_amount
+        self.numbers = [8, 9, 10, 11, 12]
+        self.payoutratio = 30

```

Comment:
I should have clarified, but this should be `self.payoutratio = 29`, with 29 for AllSmall and 149 for AllOrNothingAtAll.  This reflects the 30/30/150 for 1, i.e. 29/29/149 to 1.